compiler/src/iree/compiler/GlobalOptimization/PropagateLinalgTranspose.cpp

```diff
@@ -657,22 +692,57 @@ class SinkTransposeThroughUnaryElementwiseInput
   LogicalResult matchAndRewrite(linalg::GenericOp genericOp,
                                 PatternRewriter &rewriter) const override {
     if (!IREE::Flow::isNonNullAndOutsideDispatch(genericOp)) {
-      return failure();
+      return rewriter.notifyMatchFailure(genericOp, "pre-formed dispatch");
     }
 
-    if (!isUnaryElementwiseGeneric(genericOp)) {
-      return rewriter.notifyMatchFailure(genericOp, "not unary elementwise");
+    if (!linalg::isElementwise(genericOp)) {
+      return rewriter.notifyMatchFailure(genericOp, "non-elementwise generic");
     }
 
-    auto transposeOp =
-        genericOp.getDpsInputs()[0].getDefiningOp<linalg::TransposeOp>();
-    if (!transposeOp) {
-      return rewriter.notifyMatchFailure(genericOp, "no transpose operand");
+    if (genericOp.getNumDpsInits() != 1) {
+      return rewriter.notifyMatchFailure(genericOp,
+                                         "unimplemented: multiple results");
     }
 
-    if (!transposeOp->hasOneUse()) {
```

Comment:
Do we no longer need the check? I thought that we still need it.

Reply:
This is checked for below instead because we scan all of the inputs for the transpose op (comment saying `// Skip multi-use transposes.`).